packages/vm/src/crypto.rs

```diff
@@ -174,4 +196,33 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_ed25519_verify() {
+        // Explicit hash
+        let message_hash = Sha256::new().chain(MSG).finalize();
```

Comment:
In ed25519 we never work on the hash directly. Hashing (SHA-512) is performed internally. So this is always just `message` of arbitrary length.

```suggestion
        let message = MSG;
```